interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -277,15 +280,65 @@ function($log, $cacheFactory, $q, Sample, Activity) {
       //  note: progress can be reported by returning a $promise to the caller
       // respObj.queryStatus = 'Retrieving sample activity...';
       this.rebuildHeatmapActivity(this.heatmapData.samples);
+    },
+
+    // volcano plot methods
+    getVolcanoPlotData: function() {
```

Comment:
I'm not wild about that aspect of JavaScript, either, but the only thing to be done for that is to write in a different language. 😄 

We don't really have a choice, since it is a part of the language. Assigning to a temp variable doesn't really make the problem go away, either. It's just something you need to do in some instances to work around this aspect of the language. All of the cases where it's used in this part of the code are done correctly.